lib/bgp/server/session.ex

```diff
@@ -104,14 +106,24 @@ defmodule BGP.Server.Session do
             socket: nil,
             start: nil,
             timers: %{},
-            transport: nil
+            transport: nil,
+            transport_opts: []
 
   @doc false
-  def child_spec(opts), do: %{id: __MODULE__, start: {__MODULE__, :start_link, [opts]}}
+  def child_spec({opts, _}),
+    do: %{id: make_ref(), start: {__MODULE__, :start_link, [opts]}}
+
+  def child_spec(opts),
+    do: %{id: make_ref(), start: {__MODULE__, :start_link, [opts]}}
 
   @spec check_collision(t(), OPEN.bgp_id()) :: :ok | {:error, :collision}
-  def check_collision(session, peer_bgp_id),
-    do: :gen_statem.call(session, {:check_collision, peer_bgp_id})
+  def check_collision(session, peer_bgp_id) do
+    if session == self() do
+      :ok
+    else
```

Comment:
Is there a way to avoid this check? It's not clear to me why we need this.

Reply:
We need this because the session looks for a registered process and it may be itself. We can maybe move the check somewhere else but it is needed to avoid a session doing collision detection with itself.